solvation_analysis/analysis_library.py

```diff
@@ -0,0 +1,116 @@
+import pandas as pd
+import numpy as np
+
+class _Speciation:
+    """
+    A class for calculating and storing speciation information for solvents.
+    """
+
+    def __init__(self, solvation_data, n_frames, n_solutes):
+        """
+
+        Parameters
+        ----------
+        solvation_data: the solvation data frame output by Solute
+        """
+        self.solvation_data = solvation_data
+        self.n_frames = n_frames
+        self.n_solutes = n_solutes
+        self.speciation, self.speciation_percent = self._compute_speciation()
+
+        # self.solvation_frames = solvation_data.solvation_frames
+        # self.speciation_frames = self._accumulate_speciation(self.solvation_frames)
+        # self.average_speciation = self._average_speciation(
+        #     self.speciation_frames, solvation_data.solute_number, solvation_data.frame_number
+        # )
+        # self.res_names = self._single_speciation_frame(
+        #     self.solvation_frames[0], return_res_names=True
+        # )
+
+    def _compute_speciation(self):
+        counts = self.solvation_data.groupby(["frame", "solvated_atom", "res_name"]).count()["res_id"]
```

Comment:
What if I want to group by some other kwarg? like atomname or Bfactor or whatever. The use of explicit orderings based on a certain attribute is discouraged, the use of AtomGroup fixes this by allowing any selection criteria. I don't think this can be changed at this stage but something to note for down the track.

Reply:
Definitely worth thinking about what an AtomGroup based data structure would look like. @hmacdope do you think you could open an issue on this? We can continue discussion there in the future since I don't think there is time now.

I'd make one but I want to make sure your suggestion is conveyed accurately.

Reply:
Now raised as #32

Reply:
Great, thank you! Going to resolve this now.